Sources/Features/Breadcrumb/BacktraceBreadcrumbFile.swift

```diff
@@ -31,54 +31,39 @@ enum BacktraceBreadcrumbFileError: Error {
 
     func addBreadcrumb(_ breadcrumb: [String: Any]) -> Bool {
         do {
-            // Serialize breadcrumb: [String: Any] into Data
             let breadcrumbJsonData = try JSONSerialization.data(withJSONObject: breadcrumb)
-            // Serialize Data into a JSON string
             guard let breadcrumbJsonString = String(data: breadcrumbJsonData, encoding: .utf8) else {
                 BacktraceLogger.warning("Error when converting breadcrumb to string")
                 return false
             }
-            // Calculate the size of the breadcrumb and add it to queue
             let breadcrumbSize = breadcrumbJsonData.count
             // Check if breadcrumb size is larger than the maximum specified
             if breadcrumbSize > maximumIndividualBreadcrumbSize {
                 BacktraceLogger.warning(
                     "Discarding breadcrumb that was larger than the maximum specified (\(maximumIndividualBreadcrumbSize).")
                 return false
             }
-            // Store breadcrumb Json String and size in Dictionary [String : Any]
-            let queueBreadcrumb = ["breadcrumbJson": breadcrumbJsonString, "size": breadcrumbSize] as [String : Any]
-            // Queue breacrumb
+            let queueBreadcrumb = BreadcrumbRecord(size: breadcrumbSize, json: breadcrumbJsonString)
             queue.enqueue(queueBreadcrumb)
-            // Iterate over the queue from newest to oldest breadcrumb and build an array of encoded strings
             _ = dispatchQueue.sync {
                 let queuedBreadcrumbs = queue.allElements()
                 var breadcrumbsArray = [String]()
                 var size = 0
                 for index in (0..<queue.count).reversed() {
-                    guard let queueBreadcrumb = queuedBreadcrumbs[index] as? [String: Any] else {
-                        BacktraceLogger.warning("Error when fetching breacrumbs from queue")
-                        return false
-                    }
-                    guard let breadcrumbSize = queueBreadcrumb["size"] as? Int else {
-                        BacktraceLogger.warning("Error when adding breadcrumbSize to array")
-                        return false
-                    }
+                    let queueBreadcrumb = queuedBreadcrumbs[index]
```

Comment:
We receive index out of bounds when we access breadcrumbs with an invalid index, right? or will it return null?

Reply:
the way the code is laid out inside DispatchQueue it shouldn't produce an invalid index.
To be extra sure I'm going to add a method that returns an optional element at index and guard that.

Reply:
it shouldn't but I'll guard it